pkg/controller/util.go

```diff
@@ -933,54 +941,106 @@ func pvcFromStorage(client client.Client, recorder record.EventRecorder, log log
-			if err != nil {
-				log.V(1).Info("Cannot set accessMode for new pvc", "namespace", dv.Namespace, "name", dv.Name, "Error", err)
-				recorder.Eventf(dv, v1.EventTypeWarning, ErrClaimNotValid,
-					fmt.Sprintf("DataVolume.storage spec is missing accessMode and cannot get access mode from StorageProfile %s", getName(storageClass)))
-				return nil, err
-			}
-			pvcSpec.AccessModes = append(pvcSpec.AccessModes, accessModes...)
-		} else if pvcSpec.VolumeMode == nil || *pvcSpec.VolumeMode == "" {
-			volumeMode, err := getDefaultVolumeMode(client, storageClass, pvcSpec.AccessModes)
-			if err != nil {
-				return nil, err
-			}
-			pvcSpec.VolumeMode = volumeMode
+		return nil
+	}
+	pvcSpec.StorageClassName = &storageClass.Name
+	// given storageClass we can apply defaults if needed
+	if (pvcSpec.VolumeMode == nil || *pvcSpec.VolumeMode == "") && (len(pvcSpec.AccessModes) == 0) {
+		accessModes, volumeMode, err := getDefaultVolumeAndAccessMode(client, storageClass)
+		if err != nil {
+			log.V(1).Info("Cannot set accessMode and volumeMode for new pvc", "namespace", dv.Namespace, "name", dv.Name, "Error", err)
+			recorder.Eventf(dv, v1.EventTypeWarning, ErrClaimNotValid,
+				fmt.Sprintf("DataVolume.storage spec is missing accessMode and volumeMode, cannot get access mode from StorageProfile %s", getName(storageClass)))
+			return err
+		}
+		pvcSpec.AccessModes = append(pvcSpec.AccessModes, accessModes...)
+		pvcSpec.VolumeMode = volumeMode
+	} else if len(pvcSpec.AccessModes) == 0 {
+		accessModes, err := getDefaultAccessModes(client, storageClass, pvcSpec.VolumeMode)
+		if err != nil {
+			log.V(1).Info("Cannot set accessMode for new pvc", "namespace", dv.Namespace, "name", dv.Name, "Error", err)
+			recorder.Eventf(dv, v1.EventTypeWarning, ErrClaimNotValid,
+				fmt.Sprintf("DataVolume.storage spec is missing accessMode and cannot get access mode from StorageProfile %s", getName(storageClass)))
+			return err
 		}
+		pvcSpec.AccessModes = append(pvcSpec.AccessModes, accessModes...)
+	} else if pvcSpec.VolumeMode == nil || *pvcSpec.VolumeMode == "" {
+		volumeMode, err := getDefaultVolumeMode(client, storageClass, pvcSpec.AccessModes)
+		if err != nil {
+			return err
+		}
+		pvcSpec.VolumeMode = volumeMode
 	}
+	return nil
+}
 
-	requestedVolumeSize, err := volumeSize(client, storage, pvcSpec.VolumeMode)
-	if err != nil {
-		return nil, err
-	}
+func setStorageByHint(dv *cdiv1.DataVolume, pvcSpec *v1.PersistentVolumeClaimSpec) error {
 	if pvcSpec.Resources.Requests == nil {
 		pvcSpec.Resources.Requests = v1.ResourceList{}
 	}
-	pvcSpec.Resources.Requests[v1.ResourceStorage] = *requestedVolumeSize
-
-	return pvcSpec, nil
+	if storageHint := dv.Annotations[annResourceStorageHint]; storageHint != "" {
+		size, err := strconv.Atoi(storageHint)
+		if err != nil {
+			return err
+		}
+		//FIXME: configurable factor/delta for reservation
+		storageResizeFactor := 1.25
```

Comment:
I do not understand this, why not just use 1 and leave the size inflation to fsOverhead logic?

Reply:
So why not just use 1. User wants to import and have usable OS. If I am not mistaken - virtual images already have free space in it. If not, the user can always resize. 

So I think - if you want to have a simple and quick import you just do it with factor = 1. If you want some advanced logic then you do this logic in import/migrate tools, and the tool can set correct Request Size.  

So I do not know, I would like to see a real scenario - a real case why this might be needed.

Reply:
Convinced, fixed, thanks!